File: frontend/nobet_app/lib/models/notification.dart
class AppNotification {
  final String id;
  final String userId;
  final String title;
  final String message;
  final bool isRead;
  final String createdAt;

  const AppNotification({
    required this.id,
    required this.userId,
    required this.title,
    required this.message,
    required this.isRead,
    required this.createdAt,
  });

  factory AppNotification.fromJson(Map<String, dynamic> j) => AppNotification(
        id: j['id'],
        userId: j['userId'],
        title: j['title'] ?? '',
        message: j['message'] ?? '',
        isRead: j['isRead'] ?? false,
        createdAt: j['createdAt'] ?? '',
      );
}

File: backend/src/NobetSistemi.Domain/Entities/Notification.cs
namespace NobetSistemi.Domain.Entities;

public class Notification
{
    public Guid Id { get; set; }
    public Guid? GroupId { get; set; }
    public Group? Group { get; set; }
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;
    public string Title { get; set; } = string.Empty;
    public string Message { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
